### PR TITLE
Regular cleanup

### PR DESCRIPTION
**Pull request by smorimoto**

### CODE REVIEW

**Comment by smorimoto** [comment]
The build field updates in the opam file are from dune 2.9, which is the default.

File: dune-workspace.dev [hunk 1]
```diff
@@ -1,16 +0,0 @@
-(lang dune 1.11)
```
Comment (hhugo) [inline dune-workspace.dev:1]: I would like to keep that file

File: dune [hunk 2]
```diff
@@ -1,10 +1,3 @@
-(vendored_dirs vendor)
-
-(env
```
Comment (hhugo) [inline dune:3]: I would like to keep this field

File: dune-workspace.dev (hunk 1 above)
Comment (smorimoto) [reply dune-workspace.dev:1]: I kept it, and updated it.

File: dune (hunk 2 above)
Comment (smorimoto) [reply dune:3]: I restored just it.

File: dune-project [hunk 3]
```diff
@@ -1,14 +1,13 @@
-(lang dune 2.8)
+(lang dune 2.9)
 (name js_of_ocaml)
 
 (generate_opam_files true)
 
 (authors "Ocsigen team <dev@ocsigen.org>")
 (maintainers "Ocsigen team <dev@ocsigen.org>")
 (source (github ocsigen/js_of_ocaml))
-(bug_reports "https://github.com/ocsigen/js_of_ocaml/issues")
```
Comment (hhugo) [inline dune-project:9]: Why is bug_reports gone ?

Comment (smorimoto) [reply dune-project:9]: If github is specified in the source stanza, it's generated automatically.

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -61,8 +61,6 @@ jobs:
           opam-depext: ${{ !matrix.skip-test }}
           opam-depext-flags: --with-test
 
-      - run: opam pin add ppx_expect https://github.com/hhugo/ppx_expect.git#v0.14.1+winfix --no-action
```
Comment (smorimoto) [inline .github/workflows/build.yml:64]: Now that 0.14.2 has been released, we can safely remove this line!

**Comment by smorimoto** [comment]
I think this is now ready for merging.

**Review by hhugo — APPROVED** [review]
LGTM

**Pull request merged**